compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala

```diff
@@ -3899,8 +3915,13 @@ class JSCodeGen()(using genCtx: Context) {
 
       (f, true)
     } else*/ {
-      val f = js.Select(qual, encodeClassName(sym.owner),
-          encodeFieldSym(sym))(toIRType(sym.info))
+      val f =
+        if sym.is(JavaStatic) then
+          js.SelectStatic(encodeClassName(sym.owner),
+            encodeFieldSym(sym))(toIRType(sym.info))
+        else
+          js.Select(qual, encodeClassName(sym.owner),
+            encodeFieldSym(sym))(toIRType(sym.info))
```

Comment:
Consider computing the `className`, `fieldIdent` and `irType` in three `val`s before the `if`. That way it's clear that they're the same in both branches.

Reply:
done